subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/SwiftProjectInitDescriptor.java

```diff
@@ -0,0 +1,132 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.buildinit.plugins.internal;
+
+import com.google.common.collect.Sets;
+import org.gradle.buildinit.plugins.internal.modifiers.BuildInitTestFramework;
+import org.gradle.buildinit.plugins.internal.modifiers.Language;
+import org.gradle.internal.os.OperatingSystem;
+import org.gradle.nativeplatform.platform.internal.DefaultNativePlatform;
+
+import java.util.Set;
+
+import static org.gradle.buildinit.plugins.internal.ModuleNameBuilder.toModuleName;
+
+public abstract class SwiftProjectInitDescriptor extends LanguageLibraryProjectInitDescriptor {
+    private final TemplateOperationFactory templateOperationFactory;
+
+    public SwiftProjectInitDescriptor(TemplateOperationFactory templateOperationFactory) {
+        this.templateOperationFactory = templateOperationFactory;
+    }
+
+    @Override
+    public Language getLanguage() {
+        return Language.SWIFT;
+    }
+
+    @Override
+    public void generate(InitSettings settings, BuildScriptBuilder buildScriptBuilder, TemplateFactory templateFactory) {
+        buildScriptBuilder
+            .fileComment("This generated file contains a sample Swift project to get you started.")
+            .fileComment("For more details take a look at the Building Swift applications and libraries chapter in the Gradle");
+        configureBuildScript(settings, buildScriptBuilder);
+
+        TemplateOperation sourceTemplate = sourceTemplateOperation(settings);
+        TemplateOperation testSourceTemplate = testTemplateOperation(settings);
+
+        if (OperatingSystem.current().isLinux() || OperatingSystem.current().isWindows()) {
+            TemplateOperation testEntryPointTemplate = testEntryPointTemplateOperation(settings);
+            templateFactory.whenNoSourcesAvailable(sourceTemplate, testSourceTemplate, testEntryPointTemplate).generate();
+        } else {
+            templateFactory.whenNoSourcesAvailable(sourceTemplate, testSourceTemplate).generate();
+        }
+    }
+
+    @Override
+    public Set<BuildInitTestFramework> getTestFrameworks() {
+        return Sets.newHashSet();
+    }
+
+    @Override
+    public BuildInitTestFramework getDefaultTestFramework() {
+        return null;
+    }
+
+    protected abstract TemplateOperation sourceTemplateOperation(InitSettings settings);
+
+    protected abstract TemplateOperation testTemplateOperation(InitSettings settings);
+
+    protected abstract TemplateOperation testEntryPointTemplateOperation(InitSettings settings);
+
+    protected void configureBuildScript(InitSettings settings, BuildScriptBuilder buildScriptBuilder) {
+    }
+
+    @Override
+    public boolean supportsPackage() {
+        return false;
+    }
+
+    protected String getHostTargetMachineDefinition() {
+        DefaultNativePlatform host = DefaultNativePlatform.host();
+        if (host.getOperatingSystem().isWindows()) {
```

Comment:
I don't think this is ever hit?

Can we just assert that this isn't Windows?

Reply:
I removed it and added an assertion.